src/main/kotlin/io/icure/md/client/apis/impl/AuthenticationApiImpl.kt

```diff
@@ -0,0 +1,225 @@
+import reactor.netty.ByteBufFlux
+import reactor.netty.http.client.HttpClient
+import java.io.IOException
+import java.security.KeyPair
+import java.util.*
+import kotlin.time.DurationUnit
+import kotlin.time.ExperimentalTime
+import kotlin.time.toDuration
+import io.icure.kraken.client.models.decrypted.PatientDto as DecryptedPatientDto
+
+@ExperimentalStdlibApi
+@FlowPreview
+@ExperimentalCoroutinesApi
+@ExperimentalTime
+@ExperimentalUnsignedTypes
+class AuthenticationApiImpl(
+    private val iCureUrlPath: String,
+    private val authServerUrl: String,
+    private val authProcessId: String,
+    private val defaultLanguage: String
+) : AuthenticationApi {
+
+    companion object {
+        fun fromApi(api: MedTechApi) = AuthenticationApiImpl(
+            iCureUrlPath = api.iCureUrlPath,
+            authServerUrl = api.authServerUrl!!,
+            authProcessId = api.authProcessId!!,
+            defaultLanguage = api.defaultLanguage
+        )
+    }
+
+    private val objectMapper: ObjectMapper = ObjectMapper()
+        .registerModule(KotlinModule())
+        .registerModule(JavaTimeModule()).apply {
+            setSerializationInclusion(JsonInclude.Include.NON_NULL)
+            configure(DeserializationFeature.ACCEPT_SINGLE_VALUE_AS_ARRAY, true)
+            configure(JsonReadFeature.ALLOW_UNESCAPED_CONTROL_CHARS.mappedFeature(), true)
+        }
+
+    private fun client() = HttpClient
+        .create()
+        .headers { h ->
+            h.set("Content-Type", "application/json")
+        }
+
+    override suspend fun startAuthentication(
+        healthcareProfessionalId: String,
+        firstName: String,
+        lastName: String,
+        email: String,
+        recaptcha: String,
+        mobilePhone: String?
+    ): AuthenticationProcess? {
+        val requestId: String = UUID.randomUUID().toString()
+
+        val params = mapOf(
+            "g-recaptcha-response" to recaptcha,
+            "firstName" to firstName,
+            "lastName" to lastName,
+            "from" to email,
+            "mobilePhone" to mobilePhone,
+            "hcpId" to healthcareProfessionalId
+        )
+
+        val body = objectMapper.writeValueAsString(params)
+
+        val response = client().post()
+            .uri("${authServerUrl}/process/${authProcessId}/$requestId")
+            .send(ByteBufFlux.fromString(Mono.just(body)))
+            .response()
+            .awaitFirstOrNull()
+
+        return response?.takeIf { it.status().code() < 400 }?.let { AuthenticationProcess(requestId, email) }
+    }
+
+    override suspend fun completeAuthentication(
+        process: AuthenticationProcess,
+        validationCode: String,
+        patientKeyPair: KeyPair,
+        tokenAndKeyPairProvider: (String, String) -> Triple<String, String, String>?
+    ): AuthenticationResult {
+        val response = client().get()
+            .uri("${authServerUrl}/process/validate/${process.processId}-$validationCode")
```

Comment:
In AuthenticationRequest, rename field processId by requestId, to not mix with the authProcessId provided in the previous method